vhdl_lang/src/syntax/design_unit.rs

```diff
@@ -933,4 +933,34 @@ end entity y;
         assert_eq!(tok.kind, Context);
         assert_eq!(tok.pos, code.s1("context").pos());
     }
+
+    //    #[test]
+    //    fn parse_architecture_body_with_package_decl_and_body() {
+    //        let (code, design_file) = parse_ok(
+    //            "
+    //architecture arch of ent is
+    //  package my_pkg is
+    //      -- ...
+    //   end my_pkg;
+    //  package body my_pkg is
+    //      -- ...
+    //   end package body;
+    //begin
+    //end arch;
+    //",
+    //        );
+    //
+    //assert_eq!(
+    //    code.with_stream_no_diagnostics(parse_architecture_body()),
+    //        ArchitectureBody {
+    //            span: code.token_span(),
+    //            context_clause: ContextClause::default(),
+    //            ident: code.s1("arch").decl_ident(),
+    //            entity_name: None, // TODO
```

Comment:
This is just the name of the entity, i.e. `ent` in this example. You should simply be able to get a reference to that using `WithRef::new(code.s1("ent").ident())`